executor/metrics_reader.go

```diff
@@ -349,3 +390,17 @@ func (e *MetricsSummaryByLabelRetriever) retrieve(ctx context.Context, sctx sess
 	}
 	return totalRows, nil
 }
+
+// GetInstanceMap get clusterInfoMap and set it in InstanceAddrCache.
+func GetInstanceMap(sctx sessionctx.Context) (map[string]string, error) {
+	clusterInfo := make(map[string]string)
+	serversInfo, err := infoschema.GetClusterServerInfo(sctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, v := range serversInfo {
+		clusterInfo[v.StatusAddr] = v.Address
+	}
+	sctx.GetSessionVars().InstanceAddrCache = clusterInfo
```

Comment:
I think we shouldn't assign the `sctx.GetSessionVars().InstanceAddrCache = clusterInfo` here. It makes the `GetInstanceMap` violate it's semantic.